up42/utils.py

```diff
@@ -148,13 +148,15 @@ def download_gcs_not_unpack(
         output_directory: The file output directory, defaults to the current working
             directory.
     """
-    if ".tgz" in download_url:
-        file_ending = "tgz"
-    elif ".zip" in download_url:
-        file_ending = "zip"
-    else:
-        ValueError("To be downloaded file is not a TGZ/TAR or ZIP archive.")
-    out_fp = Path().joinpath(output_directory, f"output.{file_ending}")
+    parsed_url = urlparse(download_url)
+    extension = Path(parsed_url.path).suffix
+    try:
+        file_name = parse_qs(parsed_url.query)["response-content-disposition"][0].split(
+            "filename="
+        )[1]
+        out_fp = Path().joinpath(output_directory, f"{file_name}{extension}")
+    except (IndexError, KeyError):
+        out_fp = Path().joinpath(output_directory, f"output{extension}")
```

Comment:
Do we keep output if the query has not response-content-disposition? Is there any other value that can fit here?

Reply:
The query string in a URL follows the format key1=value1&key2=value2&..., where keys and values are URL-encoded. 
I could be wrong, but in future if we change our infra provider to say S3 buckets, it could be possible that it's signed URL does not contain `response-content-disposition`.

If at all happens in future, it's probably worth to have this except block so we don't break download.